sequencer/src/auction.rs

```diff
@@ -0,0 +1,231 @@
+            .fixed_size_field("account", &self.account.to_fixed_bytes())
+            .fixed_size_field("gas_price", &self.gas_price.to_fixed_bytes())
+            .fixed_size_field("bid_amount", &self.bid_amount.to_fixed_bytes())
+            .var_size_field("url", self.url.as_str().as_ref())
+            .u64_field("view", self.view.u64())
+            .var_size_field("bundle", &bincode::serialize(&self.bundle).unwrap());
+        comm.finalize()
+    }
+}
+
+impl BidTxBody {
+    /// Sign Tx
+    pub fn sign(&self, key: &EthKeyPair) -> Result<Signature, SigningError> {
+        FeeAccount::sign_builder_message(key, self.commit().as_ref())
+    }
+    /// Get account submitting the bid
+    pub fn account(&self) -> FeeAccount {
+        self.account
+    }
+    /// Get amount of bid
+    pub fn amount(&self) -> FeeAmount {
+        self.bid_amount
+    }
+}
+
+impl Default for BidTxBody {
+    fn default() -> Self {
+        let key = FeeAccount::test_key_pair();
+        let nsid = NamespaceId::from(999);
+        Self {
+            // TODO url will be builder_url, needs to be passed in from somewhere
+            url: Url::from_str("htts://sequencer:3939").unwrap(),
+            account: key.fee_account(),
+            public_key: FeeAccount::default(),
+            gas_price: FeeAmount::default(),
+            bid_amount: FeeAmount::default(),
+            view: ViewNumber::genesis(),
+            bundle: vec![nsid],
+        }
+    }
+}
+impl Default for BidTx {
+    fn default() -> Self {
+        let body = BidTxBody::default();
+        let key = FeeAccount::test_key_pair();
+        let signature = FeeAccount::sign_builder_message(&key, body.commit().as_ref()).unwrap();
+        Self { signature, body }
+    }
+}
+
+#[derive(Error, Debug, Eq, PartialEq)]
+/// Failure cases of transaction execution
+pub enum ExecutionError {
+    #[error("Invalid Signature")]
+    InvalidSignature,
+    #[error("Invalid Phase")]
+    InvalidPhase,
+    #[error("FeeError: {0}")]
+    FeeError(FeeError),
+    #[error("Could not resolve ChainConfig")]
+    UnresolvableChainConfig,
+}
+
+impl From<FeeError> for ExecutionError {
+    fn from(e: FeeError) -> Self {
+        Self::FeeError(e)
+    }
+}
+
+// TODO consider moving common functionality to trait.
+impl BidTx {
+    /// Executes `BidTx`.
+    /// * verify signature
+    /// * charge fee
+    /// * store state (maybe not for JIT)
+    // The rational behind the `Err` is to provide not only what
+    // failed, but for which variant. The entire Tx is probably
+    // overkill, but we can narrow down how much we want to know about
+    // Failed Tx in the future. Maybe we just want its name.
+    pub fn execute(
+        &self,
+        state: &mut ValidatedState,
+    ) -> Result<(), (ExecutionError, FullNetworkTx)> {
+        self.verify()
+            .map_err(|e| (e, FullNetworkTx::Bid(self.clone())))?;
+
+        // TODO review when this actually occurs in JIT auction
+        // charge the bid
```

Comment:
This should be correct for JIT assuming that only winning bids are being sent to the sequencer, which is my reading of the diagram: https://www.notion.so/espressosys/Yet-Another-Marketplace-Spec-WIP-c09873314c8241e4b4e6463594d44bb6#7a316e557df44de7a834f82182d2b7fc

Reply:
Per [discussion](https://espresso.zulipchat.com/#narrow/stream/440749-Marketplace/topic/BidTx.20Type/near/448843471), It appears that this is correct.